main.go

```diff
@@ -1,5 +1,58 @@
 package main
 
+import (
+	"context"
+	"log"
+	"os"
+
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/config"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/db"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/forman"
+	"github.com/red-hat-storage/managed-fusion-fleet-reconciler/pkg/reconciler"
+	"go.uber.org/zap"
+)
 
 func main() {
+	//logger := logr.Logger{}
+	ctx := context.Background()
+	configFilePath := os.Getenv("CONFIG_FILE_PATH")
+	if configFilePath == "" {
+		log.Fatal("CONFIG_FILE_PATH environment variable not set")
+	}
+	// parse configuration from yaml file
+	conf, err := config.LoadAndValidateConfig(configFilePath)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	zapConfig := zap.Config{
+		Encoding:         "json",
+		EncoderConfig:    zap.NewProductionEncoderConfig(),
+		OutputPaths:      []string{"stdout"},
+		ErrorOutputPaths: []string{"stderr"},
+	}
+	logger, err := zapConfig.Build()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	dbInstance, err := db.NewClient(ctx, &conf.DB)
+	if err != nil {
+		logger.Fatal(err.Error())
+	}
+	defer dbInstance.Close()
+
+	reqChan := forman.GoForman(logger, dbInstance, conf.Reconcile.Concurrency, reconciler.Reconcile)
+
+	go func() {
+		if err := dbInstance.LoadExistingProviderClusters(ctx, reqChan); err != nil {
+			logger.Fatal(err.Error())
+		}
+	}()
```

Comment:
Why are we running this as a go function?

Reply:
- just so I understand this correctly, a send on a channel is happening in this func and forced you to run this concurrently?
- if that's the case, we need to change the name of the method

Reply:
if we dont run it in go routine we need to wait until all the work is assigned before listening for the notifications, we might end up missing updates on the DB.

Reply:
@leelavg what is your suggestion

Reply:
One, I don't have any issue with implementation. It's just that I couldn't think of a better name for that function, `Load.. ` implies sync however it isn't.

Bottomline, pls give another thought if nothing strikes pls disregard my comment.

Reply:
> if we dont run it in go routine we need to wait until all the work is assigned before listening for the notifications, we might end up missing updates on the DB.

I don't see that happening. 
We have 3 responsibilities that need to run in parallel:
1. listen to notifications -> should run as part of the main goroutine
2. accept work and distribute to workers -> Forman 
3. reconcile -> worker

So as of above, there is no need for a forth goroutine